scripts/generate-index-json.ts

```diff
@@ -0,0 +1,174 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// Generates a list of applications registered by organisations from the separate
+// files listed under the index directory.
+
+import fs from 'fs/promises';
+import path from 'path';
+import { Octokit } from '@octokit/rest';
+import fetch from 'node-fetch';
+import colours from 'ansi-colors';
+
+import type {
+    OrgIndex,
+    AppIndex,
+    validOrgKinds,
+    Organization,
+    Application,
+} from '../site/src/schema';
+
+type ParsedOrgFile = {
+    /** GitHub user ID. */
+    id: string;
+    orgIndex: OrgIndex;
+};
+
+const partnerIds: string[] = [];
+
+function notUndefined<T>(value: T | undefined): value is T {
+    return value !== undefined;
+}
+
+function initialiseGitHubApi() {
+    let authToken: string | undefined = undefined;
+    const args = process.argv.slice(2);
+    if (args[0] === '--authToken' && args[1]) {
+        authToken = args[1];
+    } else {
+        authToken = process.env.AUTH_TOKEN;
+    }
+
+    if (!authToken) {
+        console.warn(
+            colours.yellow(
+                'No auth token was provided, so you may encounter rate limit issues when calling the GitHub API.',
+            ),
+        );
+        console.warn(
+            colours.yellow(
+                'You can provide an auth token either through the "--authToken" parameter or by setting the "AUTH_TOKEN" environment variable.\n',
+            ),
+        );
```

Comment:
Should throw, IMO. This is a CI script, all warnings should be treated as errors